### PR TITLE
feat!: Split up CI workflow per product

### PR DESCRIPTION
**Pull request by Techassi**
Part of https://github.com/stackabletech/issues/issues/595

This PR splits up the `dev` workflow so that each individual product image can be built instead of building all products at once. This allows us to trigger workflows on changes in specific paths. Additionally, it is now possible to split up the conf.py file, see #732 
 
| Before | After |
| -- | -- |
| image | image |

```[tasklist]
- [ ] Changes are OpenShift compatible
- [ ] All added packages (via microdnf or otherwise) have a comment on why they are added
- [ ] Things not downloaded from Red Hat repositories should be mirrored in the Stackable repository and downloaded from there
- [ ] All packages should have (if available) signatures/hashes verified
- [ ] Add an entry to the CHANGELOG.md file
- [ ] Integration tests ran successfully
```

### CODE REVIEW

File: .github/workflows/dev_airflow.yaml [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+          - { name: "ubicloud-standard-8-arm", arch: "arm64" }
+        versions: ${{ fromJson(needs.generate_matrix.outputs.versions) }}
+    steps:
+      - name: Checkout Repository
+        uses: actions/checkout@v4
+      - name: Build Product Image
+        id: build
+        uses: ./.github/actions/build
+        with:
+          product-name: ${{ env.PRODUCT_NAME }} # todo: how to handle workflow_dispatch (use product input)
```
Comment (lfrancke) [inline .github/workflows/dev_airflow.yaml:55]: Leftover

File: .github/workflows/dev_airflow.yaml [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+    needs: [generate_matrix, build]
+    permissions:
+      id-token: write
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        versions: ${{ fromJson(needs.generate_matrix.outputs.versions) }}
+    steps:
+      - name: Checkout Repository
+        uses: actions/checkout@v4
```
Comment (lfrancke) [inline .github/workflows/dev_airflow.yaml:77]: Needs to be pinned

File: .github/workflows/dev_airflow.yaml [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+      - name: Build Product Manifest List
+        uses: ./.github/actions/publish-manifest
+        with:
+          product: ${{ env.PRODUCT_NAME }} # todo: how to handle workflow_dispatch (use product input)
```
Comment (lfrancke) [inline .github/workflows/dev_airflow.yaml:81]: See above

**Review by lfrancke — APPROVED** [review]

File: .github/actions/build/action.yml [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+inputs:
+  product-name:
+    description: The name of the product to build via bake (directory name)
+    required: true
+  product-version:
+    description: The version of the product to build via bake
+    required: true
+  image-tools-version:
+    description: The Stackable image-tools version
+    default: 0.0.8
```
[the remark this answers is not in the document]
Comment (NickLarsenNZ) [reply .github/actions/build/action.yml:13]: The upgrade will come in with https://github.com/stackabletech/docker-images/pull/735, so we won't upgrade it in this PR (it is running a recent version necessary for these changes).

File: .github/workflows/dev_airflow.yaml [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+
+env:
+  PRODUCT_NAME: airflow
+
+on:
+  workflow_dispatch:
+  schedule:
+    - cron: '0 1 1/2 * *' # https://crontab.guru/#0_1_1/2_*_*
+  push:
+    branches:
```
[the remark this answers is not in the document]
Comment (NickLarsenNZ) [reply .github/workflows/dev_airflow.yaml:12]: As discussed, we don't think a merge queue would be super useful here and will only delay the merge (the reason this PR exists is to speed up the process).

File: .github/workflows/dev_airflow.yaml (hunk 3 above)
Comment (NickLarsenNZ) [reply .github/workflows/dev_airflow.yaml]: Removed in 8c9844e672be274ab11c3ba47b01b943e43afbe8

File: .github/workflows/dev_airflow.yaml (hunk 2 above)
Comment (NickLarsenNZ) [reply .github/workflows/dev_airflow.yaml]: Pinned in b4e9c2692cee779932d324c7215f3e9091424def

File: .github/workflows/dev_airflow.yaml (hunk 1 above)
Comment (NickLarsenNZ) [reply .github/workflows/dev_airflow.yaml]: Removed in 8c9844e672be274ab11c3ba47b01b943e43afbe8

File: .github/actions/shard/action.yml [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+    description: The name of the product to build via bake (directory name)
+    required: true
+outputs:
+  versions:
+    description: A list of product versions
+    value: ${{ steps.generate_shards.outputs.VERSIONS }}
+runs:
+  using: composite
+  steps:
+    - uses: actions/setup-python@v5
```
[the remark this answers is not in the document]
Comment (NickLarsenNZ) [reply .github/actions/shard/action.yml]: Pinned in 420b5716ac42186f530fa3c6944fbb84072477e4

File: .github/actions/publish-image/action.yml [hunk 7]
```diff
@@ -0,0 +1,120 @@
...
+        # push`, because signing by tag is deprecated and will be removed from
+        # cosign in the future
+        DIGEST=$(echo "$PUSH_OUTPUT" | awk "/: digest: sha256:[a-f0-9]{64} size: [0-9]+$/ { print \$3 }")
+        echo "DIGEST=$DIGEST" >> $GITHUB_ENV
+        # Refer to image via its digest (docker.stackable.tech/stackable/airflow@sha256:0a1b2c...)
+        # This generates a signature and publishes it to the registry, next to the image
+        # Uses the keyless signing flow with Github Actions as identity provider
+        cosign sign -y "$IMAGE_NAME@$DIGEST"
+
+    - name: Generate SBOM for the Image
```
[the remark this answers is not in the document]
Comment (NickLarsenNZ) [reply .github/actions/publish-image/action.yml]: Done in ee9c159a7cccffa48b7c8b48d532844fe1c047e1

File: .github/actions/publish-image/action.yml [hunk 8]
```diff
@@ -0,0 +1,120 @@
...
+    description: The name of the product to publish
+    required: true
+  image-name:
+    description: This is the full image name before the tag (left of ':')
+    required: true
+  image-version:
+    description: This is the container image end tag excluding the architecture information (right of ':')
+    required: true
+  nexus-password:
+    description: The password to login to Nexus
```
[the remark this answers is not in the document]
Comment (NickLarsenNZ) [reply .github/actions/publish-image/action.yml:18]: Done in 41dbdbf87cba82f02a357cab1844c44a770076ad

**Review by lfrancke — APPROVED** [review]

**Pull request merged**